client/src/test/java/io/pravega/client/batch/impl/BatchClientImplTest.java

```diff
@@ -100,6 +109,33 @@ public void testGetSegmentsWithNullStreamCut() throws Exception {
         assertFalse(segments.hasNext());
     }
 
+    @Test(timeout = 5000)
+    public void testGetSegmentsWithMultipleSegments() throws Exception {
+
+        PravegaNodeUri location = new PravegaNodeUri("localhost", 0);
+        MockConnectionFactoryImpl connectionFactory = getMockConnectionFactory(location);
+        MockController mockController = new MockController(location.getEndpoint(), location
+                .getPort(), connectionFactory, false);
+        MockController stubbedController = spy(mockController);
```

Comment:
Since the passed object is not a real object, using a `spy()` seems unnecessary here. We can use `mockController` directly.

Reply:
The `mockController` used here is not a @Mock but a test implmentation for the controller.

Reply:
I see. :)